dmoj/executors/RUST.py

```diff
@@ -56,15 +57,42 @@ def create_files(self, problem_id, source_code, *args, **kwargs):
         with open(self._file('Cargo.toml'), 'wb') as f:
             f.write(CARGO_TOML)
 
+    shared_target = None
+
+    @classmethod
+    def get_shared_target(cls):
+        if cls.shared_target is not None:
+            return cls.shared_target
+
+        cargo_dir = os.path.expanduser('~/.cargo')
+        collisions = 0
+        while True:
+            maybe_target = cargo_dir + f'/dmoj-shared-target-{collisions}'
+            try:
+                os.mkdir(maybe_target, mode=0o775)
+            except FileExistsError:
+                pass
```

Comment:
Is the intention here to never clean up the files in `.cargo/dmoj-shared-target-...`?

Reply:
Yes, and to not collide with other judge processes when running outside Docker (hence the flocks). I guess this could use a comment though.

Reply:
Commented.